ccmlib/node.py

```diff
@@ -1891,8 +1891,10 @@ def __gather_sstables(self, datafiles=None, keyspace=None, columnfamilies=None):
         return files
 
     def __cleanup_sstables(self, keyspace, cf):
-        if keyspace != 'system_schema':
-            self.get_sstables('system_schema', '', cleanup_unsealed=True)
+        system_keyspace_names = ['system_schema', 'system']
+        if keyspace not in system_keyspace_names:
+            for system_keyspace_name in system_keyspace_names:
+                self.get_sstables(system_keyspace_name, '', cleanup_unsealed=True)
```

Comment:
it's not directly related to ccm. it's just how `sstablelevelreset` works. it tries to open sstables in system keyspaces. if we need to stick to `sstablelevelreset`, we have to clean the unsealed sstables before inviting it to look at the data directory.

Reply:
> It's obvious that this function was written with Cassandra in mind. This other existing hunk doesn't apply to Scylla to either:
> 
> ```
>             if os.path.exists(f.replace('Data.db', 'Compacted')):
>                 files.remove(f)
> ```

we can drop this bit in another change.

> 
> I think that just `ignore_unsealed=True, cleanup_unsealed=False` should do the right job (for a ScyllaNode).

i don't think so.

Reply:
> It's obvious that this function was written with Cassandra in mind. This other existing hunk doesn't apply to Scylla to either:
> 
> ```
>             if os.path.exists(f.replace('Data.db', 'Compacted')):
>                 files.remove(f)
> ```

we can drop this bit in another change.

> 
> I think that just `ignore_unsealed=True, cleanup_unsealed=False` should do the right job (for a ScyllaNode).

i don't think so.

Reply:
1. Why does the function even mess with system_schema if it was called for a user keyspace?  I guess it is configured with LCS in Cassandra. I'm not sure if it applies to scylla.
2. Touching unsealed sstables means that sstable components might be incomplete, how can we touch them?  The caller should stop the node before doing this or ensure there will be no unsealed sstables in the table, e.g. by disabling compaction and running flush.

Reply:
> we can drop this bit in another change.

CCM should support both Cassandra and Scylla.
Since Scylla has no `Compacted` component this logic won't apply to it.

Reply:
> Why does the function even mess with system_schema if it was called for a user keyspace? I guess it is configured with LCS in Cassandra. I'm not sure if it applies to scylla.

I guess in order to obtain the schema. Scylla sstable will also read system-schema tables every time it is invoked on any sstables (unless otherwise instructed).

Reply:
> 1. Why does the function even mess with system_schema if it was called for a user keyspace?  I guess it is configured with LCS in Cassandra. I'm not sure if it applies to scylla.

again, it's the behavior of `sstablelevelreset`. this function is change to adapt to its behavior of it in the same spirit of 3616251066f8a520a7eec99d507dc16128e2926f.   `sstablelevelreset` reads the system keyspaces even if we only instruct it to reset the table in a user keyspace. i tried to understand why `sstablelevelreset` is doing so, but i don't have enough bandwidth on read it. this test was inherited from the cassandra-dtest, if we don't want to apply it to scylla, i will create another PR to disable it.

> 
>     2. Touching unsealed sstables means that sstable components might be incomplete, how can we touch them?  The caller should stop the node before doing this or ensure there will be no unsealed sstables in the table, e.g. by disabling compaction and running flush.

we do run
```py
node1.flush()
cluster.stop(gently=False)
```

but i am afraid that `kill -9` won't allow scylla to finish the ongoing compaction -- i noticed that some of the missing/corrupted sstables were the destination sstables of compaction.

Reply:
> 1. Why does the function even mess with system_schema if it was called for a user keyspace?  I guess it is configured with LCS in Cassandra. I'm not sure if it applies to scylla.
> 2. Touching unsealed sstables means that sstable components might be incomplete, how can we touch them?  The caller should stop the node before doing this or ensure there will be no unsealed sstables in the table, e.g. by disabling compaction and running flush.

`test_sstablelevelreset` is stopping the node, and on most cases calling flush before that.
except for the first time, when it calls the tool before populating the data, there it's missing a flush call.

Reply:
but the failure are in places where a flush call was done, but node is being stopped with `cluster.stop(gently=False)`, so it can be kill during compaction.

should we disable compaction completely for this test ? and that's it ? i.e. for this tool it's should be part of the instructions (if it ever had one)

Reply:
None of these tools (be they C* or Scylla) work well when the node can compact while they are reading the sstables. Any test which wants to use these, should either make sure to disable compaction or stop the node when the tool is running.

Reply:
So scylla is stopped, but not gently so it may have unsealed sstables that were in the processing of being written while the node was taken down.
It should actually be OK to remove all of them, as we do it anyhow on restart.
By mistake, I thought we're resetting the level for those unsealed sstables and sstables in the system keyspaces, but I see we only call `self.get_sstables` to perform this cleanup of unsealed sstables, which is OK.
Not elegant at all, but should work.

Reply:
agreed, it is even confusing. i will create a follow-up change to explain the rationale in a comment in this function. but since it's failing in multiple places frequently. will leave it for another PR.

Reply:
> None of these tools (be they C* or Scylla) work well when the node can compact while they are reading the sstables. Any test which wants to use these, should either make sure to disable compaction or stop the node when the tool is running.

scylla is, well, stopped, in a harsh way.